CMakeLists.txt

```diff
@@ -50,6 +63,11 @@ if(MINGW)
   set(EXTRA_LIBS ${EXTRA_LIBS} ws2_32)
   set(TEST_LIBS ${TEST_LIBS} ws2_32)
 endif()
+# BSD doesn't link pthread by default, so make it explicit
+if(NOT WIN32)
+  set(EXTRA_LIBS ${EXTRA_LIBS} pthread)
+  set(TEST_LIBS ${TEST_LIBS} pthread)
+endif()
```

Comment:
I am not comfortable with this change.

For instance, threading is part of libc in FreeBSD, and you have to compile with `-pthread` (not `-lpthread`). On some platforms you also need to specify additional macros on the command-line.

CMake has a `FindThreads` function. May we could use that instead?